x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteForecastAction.java

```diff
@@ -137,26 +141,20 @@ private void deleteForecasts(SearchResponse searchResponse,
 
         executeAsyncWithOrigin(client, ML_ORIGIN, DeleteByQueryAction.INSTANCE, deleteByQueryRequest, ActionListener.wrap(
             response -> {
-                if (response.getDeleted() > 0) {
-                    logger.info("Deleted forecast(s) [{}] from job [{}]", forecastIds, jobId);
-                    listener.onResponse(new AcknowledgedResponse(true));
-                    return;
-                }
                 if (response.isTimedOut()) {
                     listener.onFailure(
                         new TimeoutException("Delete request timed out. Successfully deleted " +
-                            response.getDeleted() + " forecasts from job [" + jobId + "]"));
+                            response.getDeleted() + " forecast documents from job [" + jobId + "]"));
                     return;
                 }
-                if (response.getBulkFailures().isEmpty() && response.getSearchFailures().isEmpty()) {
+                if ((response.getBulkFailures().isEmpty() && response.getSearchFailures().isEmpty()) == false) {
+                    Tuple<RestStatus, Throwable> statusAndReason = getStatusAndReason(response);
```

Comment:
We discussed that we should ignore version conflict exceptions as they imply the document has already been deleted. Won't those be included in the bulk failures? If yes, I don't see where we filter them out.

Reply:
@dimitris-athanasiou I tested with ignore false and true. When the ignore is true, no exception is thrown due to version conflict, however, if the ignore is false, I do get an exception thrown. So, I don't think any filtering is necessary from my testing. 

I added a test that fails when that field is false (as it does not expect an exception), and passes when true so that we can be alerted through regression if this behavior changes.